src/wallet/kasWallet.js

```diff
@@ -0,0 +1,179 @@
+    }
+
+    /**
+     * Generates accounts in the KAS wallet api service with randomly generated key pairs.
+     *
+     * @param {number} numberOfAccounts The number of accounts to create.
+     * @return {Array.<string>}
+     */
+    async generate(numberOfAccounts) {
+        const addresses = []
+        for (let i = 0; i < numberOfAccounts; ++i) {
+            const account = await this.walletAPI.createAccount()
+            addresses.push(account.address)
+        }
+        return addresses
+    }
+
+    /**
+     * Get the account in KAS Wallet API Service corresponding to the address
+     *
+     * @param {string} address The address of account to query.
+     * @return {AccountCountByAccountID}
+     */
+    async getAccount(address) {
+        const account = await this.walletAPI.getAccount(address)
+
+        return account
+    }
+
+    /**
+     * Returns whether the account corresponding to the address exists
+     *
+     * @param {string} address The address of account to check existence.
+     * @return {boolean}
+     */
+    async isExisted(address) {
+        try {
+            await this.walletAPI.getAccount(address)
+            return true
+        } catch (e) {
+            if (e.code === 1061010) return false
+            throw e
+        }
+    }
+
+    /**
+     * Deletes the account that associates with the given address from KAS Wallet API service.
+     *
+     * @param {string} address An address of the account to be deleted in KAS Wallet API service.
+     * @return {boolean}
+     */
+    async remove(address) {
+        const deleted = await this.walletAPI.deleteAccount(address)
+        return deleted.status === 'deleted'
+    }
+
+    /**
+     * Signs the transaction using one key and return the transactionHash
+     *
+     * @param {string} address An address of account in KAS Wallet API Service.
+     * @param {AbstractTransaction} transaction A transaction object of caver-js. See [Klaytn Docs - Transaction](https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.transaction) for details.
+     * @return {AbstractTransaction}
+     * @see {@link https://docs.klaytn.com/bapp/sdk/caver-js/api-references/caver.transaction#class|Transaction}
+     */
+    async sign(address, transaction) {
+        // Check from address
+        transaction.from = transaction.from && transaction.from !== '0x' ? transaction.from : address
+        if (transaction.from.toLowerCase() !== address.toLowerCase()) throw new Error(`From addresses are not matched.`)
+
+        // Check accountKey in Klaytn network
+        const accountKey = await transaction.constructor._klaytnCall.getAccountKey(transaction.from)
+        const keyType =
+            accountKey.keyType === 5
+                ? transaction.type.includes('AccountUpdate')
+                    ? accountKey.key[1].keyType
+                    : accountKey.key[0].keyType
+                : accountKey.keyType
+        if (keyType > 3) {
```

Comment:
If the key is not RoleBasedKey, Do we need to support AccountKeyFail when signing the transaction?

Reply:
I don't think Caver needs to worry about what value is actually stored in the KAS Wallet API and tries to sign it. However, in case of multisig account, it is supported by KAS Wallet API, but since we do not support it, we think that it is correct to be filtered earlier, so an error is returned only in the case of AccountKeyWeightedMultiSig.